### ISSUE TITLE
Collections does not properly encode Decimal type

### ISSUE THREAD

**Issue by gabriel-domingues** [zookzook/elixir-mongodb-driver#156]
A `Decimal` field in a collection currently does not encode as a `BSON.Decimal128`, but rather as a nested map.

A field in a collection that is an Elixir "special type", i.e. `DateTime` or `BSON.ObjectId`, wants to use their custom `BSON` encoder/decoder. This is currently not working for the `Decimal` type. Namely, to put a `Decimal` as a field in a collection will get encoded as a map, not their own encoding specified in `BSON.Decimal128`.

The `dump` function on a Collection, is not catching that a Decimal is a "special type" and is encoding as a nested struct.

Here is an example, directly calling without a collection (on version 0.9.2).
```elixir
iex> Mongo.Collection.dump(:key, Decimal.new(10))
%{coef: 10, exp: 0, sign: 1}
iex> Mongo.Collection.dump(:key, Date.utc_today())
~D[2022-11-19]
```

I believe that if this is indeed so, the fix is simply guarding on the function `&ensure_nested_map/1` in `Mongo.Collection` with `Decimal`. The rest of BSON encode/decode pipeline seems to work correctly.

**Comment by zookzook** [comment]
Thanks for reporting. Please try out the current master branch. The dump and load functions are refactored to be more consistent. The next release will contain the changes.

**Comment by zookzook** [comment]
Fixed in the next coming release 1.0.0
